cmd/oras/internal/option/common.go

```diff
@@ -30,3 +34,8 @@ func (opts *Common) ApplyFlags(fs *pflag.FlagSet) {
 	fs.BoolVarP(&opts.Debug, "debug", "d", false, "debug mode")
 	fs.BoolVarP(&opts.Verbose, "verbose", "v", false, "verbose output")
 }
+
+// WithLogger sets a logger to ctx with proper logging options tuned.
+func (opts *Common) WithLogger(ctx context.Context) (context.Context, logrus.FieldLogger) {
```

Comment:
I agree something needs to be done here, but since you are passing in a context here, I would think this would be called `WithContext`

Reply:
Good idea, resolved.

This PR is pending on the merge of #825, I picked this PR out so the review work is easier.